designer/server/lib/persistence/previewPersistenceService.js

```diff
@@ -0,0 +1,28 @@
+// @flow
+import type { PersistenceService } from './persistenceService'
+const Wreck = require('@hapi/wreck')
+const config = require('../../../config')
+
+export class PreviewPersistenceService implements PersistenceService {
+  logger: any;
+  async uploadConfiguration (id: string, configuration: string) {
+    return Wreck.post(`${config.previewUrl}/publish`, {
+      payload: JSON.stringify({ id, configuration })
+    })
+  }
+
+  async copyConfiguration (configurationId: string, newName: string) {
+    const configuration = await this.getConfiguration(configurationId)
+    return this.uploadConfiguration(newName, JSON.parse(configuration).values)
+  }
+
+  async listAllConfigurations () {
+    const { payload } = await Wreck.get(`${config.previewUrl}/published`)
+    return Promise.resolve(JSON.parse(payload.toString()))
```

Comment:
you don't need to return Promise.resolve. returning a value from async is a resolve. it will also wrap the return value in a Promise for you

Reply:
cool, thanks